### ISSUE TITLE
RwLock and std::box - parking_lot does not allocate ?

### ISSUE THREAD

**Issue by luke-titley** [Amanieu/parking_lot#285]
Hey,
So i've been reading through the docs/notes and parking_lot sounds awesome. I'm using std RwLock a lot and I was surprised to learn a few months back that because pthread primitives are non TM that rust std lib has to box them. You mention in the notes that parking_lot RwLock is just one additional word of space. I just want to clarify, does this means a parking_lot RwLock doesn't not implicitly allocate?

**Comment by bjorn3** [comment]
Correct. parking_lot only needs a single `AtomicUsize` as state next to the actual value. This `AtomicUsize` can be moved together with the rest of the `RwLock` and as such doesn't need to be boxed.

https://github.com/Amanieu/parking_lot/blob/d763fcab5e1b76613b7d5f49fc4e8fbf09610b49/src/raw_rwlock.rs#L55-L58
